pkg/proxy/topology.go

```diff
@@ -52,6 +52,17 @@ func CategorizeEndpoints(endpoints []Endpoint, svcInfo ServicePort, nodeLabels m
 			return true
 		})
 
+		// if there are 0 cluster-wide endpoints, we can try to fallback to any terminating endpoints that are ready.
+		if len(clusterEndpoints) == 0 && utilfeature.DefaultFeatureGate.Enabled(features.ProxyTerminatingEndpoints) {
+			clusterEndpoints = filterEndpoints(endpoints, func(ep Endpoint) bool {
+				if ep.IsServing() && ep.IsTerminating() {
+					return true
+				}
+
```

Comment:
Will add a comment, but something we should probably decide on before Beta

Reply:
Added a comment.

Thinking about this more though, it wouldn't be super complicated for us to try two filters for fallback, the first accounting for topology and then checking all endpoints if the first one returned 0 endpoints. Should we just do that? The only downside I can think of is that we add more computation in CategorizeEndpoints cause it iterates the `endpoints` two extra times, but given it only occurs when no endpoint is ready and when topology is enabled it might not be significant.

Reply:
I'm not at all convincved it is necessary